src/runtime/node/url/index.ts

```diff
@@ -1,20 +1,41 @@
 // https://nodejs.org/api/url.html
-import type url from "node:url";
+import type nodeUrl from "node:url";
+import { ParsedUrlQuery } from "node:querystring";
 
 // TODO: https://github.com/unjs/unenv/issues/65
-export const URL = globalThis.URL as typeof url.URL;
+export const URL = globalThis.URL as typeof nodeUrl.URL;
 
-export const URLSearchParams = globalThis.URLSearchParams;
+export const URLSearchParams =
+  globalThis.URLSearchParams as typeof nodeUrl.URLSearchParams;
+
+// Output of `url.parse`
+export class Url extends URL implements nodeUrl.Url {
```

Comment:
oof.. this is brave. my understanding is that there are significant differences between how urls are parsed by `URL` and `Url`. Do you think we can get away with this? At minimum we should document that we know this is not a Node.js compliant implementation of `Url`.

Reply:
We can copy their impl ([here](https://github.com/nodejs/node/blob/main/lib/url.js)) but it will pull-in huge amount of polyfills.

I would suggest iterate over this and see if anything is reported since current state was worse (without adding Node.js specific props)

Reply:
yeah. I saw their code. I also noticed that Deno just took the code as is and embedded it. Let's go with your implementation and if it's causing headaches, we can implement this one natively in workerd for CF and reuse the same code in unenv for non-CF usage.

Reply:
Can you please add a comment into the code stating that this is could be a problem and we might need to reimplement it?